Makefile

```diff
@@ -160,6 +160,7 @@ Middlewares/Third_Party/LwIP/src/core/ipv6/nd6.c \
 Middlewares/Third_Party/LwIP/system/OS/sys_arch.c \
 Middlewares/Third_Party/LwIP/src/apps/mqtt/mqtt.c \
```

Comment:
I would add SDcard.h, SDcard. c, and any other files you created in the Makefile. Then re-run the build to make sure your code compiles.